packages/flutter/lib/src/material/button_style.dart

```diff
@@ -435,3 +435,33 @@ class _LerpShapes implements MaterialStateProperty<OutlinedBorder?> {
     return ShapeBorder.lerp(resolvedA, resolvedB, t) as OutlinedBorder?;
   }
 }
+
+/// How the icon in any `button.icon` should be placed, where `button` could be:
+///
+/// * [TextButton]
+/// * [ElevatedButton]
+/// * [OutlinedButton]
+///
+/// In buttons that have an icon, we generally tend to use a `Row` widget
+/// to be able to put multiple widgets in a horizontal way, aka `Row`.
+///
+/// Hover over enum values to be able to see more deailed description
+/// on what do they do.
```

Comment:
Avoid saying how to use the docs -- "hover" doesn't work on api.flutter.dev for example.

Reply:
Yea it makes sense 🤣